app/src/ui/lib/filter-list.tsx

```diff
@@ -112,6 +112,9 @@ interface IFilterListProps<T extends IFilterListItem> {
     event: React.KeyboardEvent<HTMLInputElement>
   ) => void
 
+  /** Called when the Enter key is pressed in field of type search */
+  readonly onEnterClicked?: (text: string) => void
```

Comment:
I could see calling this `onEnter`, `onEnterDown` (like `onKeyDown`), or `onEnterPressed`. 'Clicked' doesn't seem right for a keyboard key.

Reply:
Agreed! I'd say we use `onEnterPressed` as we do in https://github.com/desktop/desktop/blob/28a2ee45a220dc87592e5c8e4666c4a6006b4f2b/app/src/ui/cherry-pick/choose-target-branch.tsx#L95

Reply:
Thinking more about it, this event will be propagated not only when <kbd>Enter</kbd> is pressed, but also when the list is empty after filtering the results, so maybe we should rename it to something more generic?

Something like… `onEnterPressedWithoutFilteredItems`?

Reply:
For more context, enter does does check out a branch when there are results. I agree with @sergiou87, clarification in the code would be nice for future devs.